### ISSUE TITLE
Overlay taps don't close the drawer immediately

### ISSUE THREAD

**Issue by DanielMSchmidt** [react-native-community/react-native-drawer-layout#20]
This only occurs sometimes and may be an issue related to the pan responder taking the event.

**Comment by satya164** [comment]
This a component I had made earlier which dealt with this, important part is a the delay prop before activating pan responder:

```js
/* @flow */

import React, { Component } from 'react';
import {
  PanResponder,
  Animated,
  StyleSheet,
} from 'react-native';

type Offset = {
  top: number;
  left: number;
}

type DragCallback = (offset: Offset) => void;

type Props = {
  offset: Offset;
  dragEnabled: boolean;
  delay?: number;
  onDragStart?: DragCallback;
  onDragMove?: DragCallback;
  onDragEnd?: DragCallback;
  style?: any;
}

type State = {
  offset: Animated.ValueXY;
  zIndex: Animated.Value;
}

export default class Draggable extends Component<void, Props, State> {

  constructor(props: Props) {
    super(props);

    this.state = {
      offset: new Animated.ValueXY({ x: this.props.offset.left, y: this.props.offset.top }),
      zIndex: new Animated.Value(0),
    };
  }

  state: State;

  componentWillMount() {
    this._panResponder = PanResponder.create({
      onStartShouldSetPanResponder: this._shouldSetResponder,
      onStartShouldSetPanResponderCapture: this._shouldSetResponder,
      onMoveShouldSetPanResponder: () => this.props.dragEnabled && this._shouldRespond,
      onMoveShouldSetPanResponderCapture: () => this.props.dragEnabled && this._shouldRespond,
      onPanResponderGrant: this._handleResponderGrant,
      onPanResponderMove: this._handleResponderMove,
      onPanResponderTerminationRequest: () => true,
      onPanResponderRelease: this._handleResponderRelease,
      onPanResponderTerminate: this._handleResponderRelease,
    });
  }

  componentWillReceiveProps(nextProps: Props) {
    if (this._dragging) {
      return;
    }
    this._currentOffset = { x: 0, y: 0 };
    this.state.offset.flattenOffset();
    Animated.spring(this.state.offset, {
      toValue: { x: nextProps.offset.left, y: nextProps.offset.top },
      friction: 9,
    }).start(() =>
      this.state.zIndex.setValue(1)
    );
  }

  _timer: any;
  _shouldRespond: boolean = false;
  _dragging: boolean = false;
  _currentOffset: { x: number, y: number } = { x: 0, y: 0 };
  _panResponder: Object;

  _shouldSetResponder = () => {
    this._clearTimer();
    this._shouldRespond = false;
    this._timer = setTimeout(() => {
      this._shouldRespond = true;
    }, typeof this.props.delay === 'number' ? this.props.delay : 0);
    return false;
  };

  _clearTimer = () => {
    if (this._timer) {
      clearTimeout(this._timer);
      this._timer = null;
    }
  };

  _handleResponderGrant = () => {
    this._dragging = true;
    this.state.zIndex.setValue(2);
    /* $FlowFixMe - this expects a number to be returned for some reason */
    this.state.offset.stopAnimation((value: { x: number, y: number }) => {
      this._currentOffset = value;
      this.state.offset.setOffset(value);
      this.state.offset.setValue({ x: 0, y: 0 });
      if (this.props.onDragStart) {
        this.props.onDragStart({
          left: value.x,
          top: value.y,
        });
      }
    });
  };

  _handleResponderMove = (evt: any, gestureState: any) => {
    this.state.offset.setValue({ x: gestureState.dx, y: gestureState.dy });
    if (this.props.onDragMove) {
      this.props.onDragMove({
        left: gestureState.dx + this._currentOffset.x,
        top: gestureState.dy + this._currentOffset.y,
      });
    }
  };

  _handleResponderRelease = (evt: any, gestureState: any) => {
    this._clearTimer();
    this._shouldRespond = false;
    this._dragging = false;
    if (this.props.onDragEnd) {
      this.props.onDragEnd({
        left: gestureState.dx + this._currentOffset.x,
        top: gestureState.dy + this._currentOffset.y,
      });
    }
  };

  render() {
    const { offset, ...rest } = this.props; // eslint-disable-line no-unused-vars

    return (
      <Animated.View
        {...rest}
        {...this._panResponder.panHandlers}
        style={[
          styles.draggable,
          {
            transform: [
              { translateX: this.state.offset.x },
              { translateY: this.state.offset.y },
            ],
            // FIXME: using zIndex causes a weird bug where the item disappears
            elevation: this.state.zIndex,
          },
          this.props.style,
        ]}
      />
    );
  }
}

const styles = StyleSheet.create({
  draggable: {
    position: 'absolute',
    top: 0,
    left: 0,
  },
});
```

**Comment by DanielMSchmidt** [comment]
Thank you for the feedback, I will try to get a fix out soon including this delay. Will ping you in the review if it's okay to you :)

**Comment by DanielMSchmidt** [comment]
This will be included in the next patch, will ship `1.1.5` now.

**Comment by DanielMSchmidt** [comment]
The fix is included in `1.1.6`
